api/envoy/extensions/load_balancing_policies/dynamic_forwarding/v3/dynamic_forwarding.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+
+package envoy.extensions.load_balancing_policies.dynamic_forwarding.v3;
+
+import "envoy/config/cluster/v3/cluster.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.load_balancing_policies.dynamic_forwarding.v3";
+option java_outer_classname = "DynamicForwardingProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/dynamic_forwarding/v3;dynamic_forwardingv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Dynamic Forwarding Load Balancing Policy]
+// [#extension: envoy.load_balancing_policies.dynamic_forwarding]
+
+// Configuration for the Dynamic Forwarding Load Balancing policy.
+//
+// This policy allows endpoint picking to be implemented in an external to Envoy
+// component. While it can work with a built in HTTP filter it is primarily intended to
+// work in conjunction with the ext_proc or WASM HTTP filters.
+//
+// The ext_proc or WASM filter must implement k8s proposal for AI gateway inferences extensions
+// https://github.com/kubernetes-sigs/gateway-api-inference-extension/tree/main/docs/proposals/004-endpoint-picker-protocol
+// to integrate with this load balancing policy.
+//
+// This policy uses selected endpoints from either request headers or request metadata.
+// Use of request headers must be enabled in the configuration and is disabled by default. If enabled, Envoy
+// uses the ``x-gateway-destination-endpoint`` header for the primary endpoint. The value must contain a single ``IP:port`` token.
+// For example ``10.0.0.5:8080`` or ``[2600:4040:5204::1574:24ae]:80``. The IPv6 address is enclosed in square brackets.
+//
+// If the header is not present Envoy uses untyped request metadata under the ``envoy.lb`` key. The metadata ``protobuf.Struct``
```

Comment:
Why use untyped metadata here?  Typed metadata seems safer.

Reply:
I think it's because the untyped metadata is used more widely and easy to read/write without the type knowledge. It's more friendly for external schedular.